### PR TITLE
Residues with the same residue ids are not merged by default

### PR DESCRIPTION
**Pull request by xiki-tempula**
Issue 1306

Fixes 

Changes made in this Pull Request:
Residues with the same residue id but different residue names are treated as different residues (not merged by default now).
 - 

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
This looks like a really good fix.  There's 2 failing tests currently, but I think that's because they used to expect the bug we fixed.  If the [desired output](https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/MDAnalysisTests/coordinates/test_gro.py#L226) of these is changed to `np.ones(25)` this should fix it.  (Previously there was 25 different resnames in the `make_Universe` used for these tests, but this got squashed as the guessed resid was 1 for all of them).

**Comment by xiki-tempula** [comment]
@richardjgowers I noticed that in these two tests some functions called string.uppercase which is not in py35 and is currently trying to fix this.

File: package/CHANGELOG [hunk 1]
```diff
@@ -19,6 +19,8 @@ mm/dd/17 utkbansal
 
 Enhancements
   * Universe now works with StringIO objects for topologies and trajectories.
+  * Residues with the same residue ids are not merged by default now
```
Comment (richardjgowers) [inline package/CHANGELOG]: Also, add your name up top to the list of authors in 16.1.  You're already in AUTHORS right?

**Comment by richardjgowers** [comment]
thank you @xiki-tempula !

**Pull request merged**